server/integration/jgroups/src/main/java/org/infinispan/server/jgroups/ManagedSocketFactory.java

```diff
@@ -80,7 +80,7 @@ private Socket register(final Socket socket) {
 
     @Override
     public ServerSocket createServerSocket(String name) throws IOException {
-        return this.register(this.factory.createServerSocket(name));
+        return this.factory.createServerSocket(name);
```

Comment:
To answer my own question, `ManagedSocketFactoryTest` expects the socket to be registered even if unbound. So I guess the problem is in `UnnamedBindingRegistry`, not here.